// frappe-js-sdk/lib/utils/axios.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.getRequestHeaders = exports.getAxiosClient = void 0;
var axios_1 = require("axios");
function getAxiosClient(appURL, useToken, token, tokenType) {
    return axios_1.default.create({
        baseURL: appURL,
        headers: getRequestHeaders(useToken, tokenType, token),
        withCredentials: true,
    });
}
exports.getAxiosClient = getAxiosClient;
function getRequestHeaders(useToken, tokenType, token) {
    if (useToken === void 0) { useToken = false; }
    var headers = {
        Accept: 'application/json',
        'Content-Type': 'application/json; charset=utf-8',
    };
    if (useToken && tokenType && token) {
        headers.Authorization = "".concat(tokenType, " ").concat(token());
    }
    // in case of browser environments
    if (typeof window !== 'undefined' && typeof document !== 'undefined') {
        if (window.location) {
            headers['X-Frappe-Site-Name'] = window.location.hostname;
        }
        if (window.csrf_token && window.csrf_token !== '{{ csrf_token }}') {
            headers['X-Frappe-CSRF-Token'] = window.csrf_token;
        }
    }
    return headers;
}
exports.getRequestHeaders = getRequestHeaders;
